### PR TITLE
Refactor & Fix BlockDownloaded

### PR DESCRIPTION
**Pull request by lautarodragan**
[PR Process](https://github.com/poetapp/documentation/blob/master/process/pr-review.md#pr-process) - [PR Review Checklist](https://github.com/poetapp/documentation/blob/master/process/pr-review.md#pr-review-checklist)

### Release
 Semantic release is enabled for this repository. Make sure you follow the right commit message convention. 
 We're using semantic-release's default — [Angular Commit Message Conventions](https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#-git-commit-guidelines).

## Description of Changes

Part of https://github.com/poetapp/node/issues/46
Fixes https://github.com/poetapp/node/issues/638

### CODE REVIEW

File: src/Messaging/Messages.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+export interface LightBlock {
+  readonly hash: string
+  readonly previousHash: string
+  readonly height: number
+}
+
+export const isBlockDownloaded = (messageContent: any): messageContent is BlockDownloaded => {
+  const { poetBlockAnchors } = messageContent
+
+  const isNotClaim = pipe(
```
Comment (kennylavender) [inline src/Messaging/Messages.ts]: This can be moved out of isBlockDownloaded's scope.

File: src/Messaging/Messages.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    isClaim,
+    not
+  )
+
+  return allPass([Array.isArray(poetBlockAnchors), !poetBlockAnchors.some(isNotClaim)])
```
Comment (kennylavender) [inline src/Messaging/Messages.ts]: allPass is supposed to take a list of predicates and have a value passed into the returned function

Comment (lautarodragan) [reply src/Messaging/Messages.ts]: Ugh you're right. Thanks.

Comment (kennylavender) [reply src/Messaging/Messages.ts]: Your welcome just finished reviewing the rest of the PR, it looks good. Just `isBlockDownloaded` that needs some work, might be worth adding some unit tests for it?

**Comment by lautarodragan** [comment]
@kennylavender can you review again?

I decided to use Joi instead, to make things more declarative. Also found out about https://github.com/poetapp/node/issues/638 and fixed it, and added some unit tests.

Wasn't able to manually test it since `npm run create-claim` is failing now.

**Pull request merged**